src/display/api.js

```diff
@@ -262,13 +258,8 @@ function getDocument(src, pdfDataRangeTransport,
   params.rangeChunkSize = params.rangeChunkSize || DEFAULT_RANGE_CHUNK_SIZE;
   params.ignoreErrors = params.stopAtErrors !== true;
 
-  if (params.disableNativeImageDecoder !== undefined) {
-    deprecated('parameter disableNativeImageDecoder, ' +
-      'use nativeImageDecoderSupport instead');
-  }
   params.nativeImageDecoderSupport = params.nativeImageDecoderSupport ||
-    (params.disableNativeImageDecoder === true ? NativeImageDecoding.NONE :
-      NativeImageDecoding.DECODE);
+                                     NativeImageDecoding.DECODE;
   if (params.nativeImageDecoderSupport !== NativeImageDecoding.DECODE &&
```

Comment:
*This could probably be considered slightly out of scope, but I have to ask.*

Do we really need to print a warning message for this *particular* API parameter, considering that we don't do that for any other ones (and doing it everywhere would require adding *a lot* of code all over the place)?

Hence I'm wondering it wouldn't suffice to replace the current code with something along the following lines?
```javascript
if (!params.nativeImageDecoderSupport ||
    !(params.nativeImageDecoderSupport === NativeImageDecoding.DECODE ||
      params.nativeImageDecoderSupport === NativeImageDecoding.NONE ||
      params.nativeImageDecoderSupport === NativeImageDecoding.DISPLAY)) {
  params.nativeImageDecoderSupport = NativeImageDecoding.DECODE;
}
```

Reply:
I would personally be fine with that, but let's do that in a different PR because I'd like to keep this one limited to only removing deprecated code. Nevertheless, this is a good point and can be picked up later!